build_scripts/ubuntu_sim_ros_melodic.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+## Bash script for setting up ROS Melodic (with Gazebo 9) development environment for PX4 on Ubuntu LTS (18.04). 
+## It installs the common dependencies for all targets (including Qt Creator)
+##
+## Installs:
+## - Common dependencies libraries and tools as defined in `ubuntu_sim_common_deps.sh`
+## - ROS Melodic (including Gazebo9)
+## - MAVROS
+
+echo "Downloading dependent script 'ubuntu_sim_common_deps.sh'"
+# Source the ubuntu_sim_common_deps.sh script directly from github
+common_deps=$(wget https://raw.githubusercontent.com/PX4/Devguide/master/build_scripts/ubuntu_sim_common_deps.sh -O -)
+wget_return_code=$?
+# If there was an error downloading the dependent script, we must warn the user and exit at this point.
+if [[ $wget_return_code -ne 0 ]]; then echo "Error downloading 'ubuntu_sim_common_deps.sh'. Sorry but I cannot proceed further :("; exit 1; fi
+# Otherwise source the downloaded script.
+. <(echo "${common_deps}")
+
+# ROS Melodic
+## Gazebo simulator dependencies
+sudo apt-get install protobuf-compiler libeigen3-dev libopencv-dev -y
+
+## ROS Gazebo: http://wiki.ros.org/melodic/Installation/Ubuntu
+## Setup keys
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+## For keyserver connection problems substitute hkp://pgp.mit.edu:80 or hkp://keyserver.ubuntu.com:80 above.
+sudo apt-get update
+## Get ROS/Gazebo
+sudo apt install ros-melodic-desktop-full -y
+## Initialize rosdep
+sudo rosdep init
+rosdep update
+## Setup environment variables
+rossource="source /opt/ros/melodic/setup.bash"
+if grep -Fxq "$rossource" ~/.bashrc; then echo ROS setup.bash already in .bashrc;
+else echo "$rossource" >> ~/.bashrc; fi
+eval $rossource
+
+## Install rosinstall and other dependencies
+sudo apt install python-rosinstall python-rosinstall-generator python-wstool build-essential -y
+
+
+
+# MAVROS: https://dev.px4.io/en/ros/mavros_installation.html
+## Install dependencies
+sudo apt-get install python-catkin-tools python-rosinstall-generator -y
+
+## Create catkin workspace
+mkdir -p ~/catkin_ws/src
+cd ~/catkin_ws
+catkin init
+wstool init src
+
+
+## Install MAVLink
```

Comment:
@TSC21 Following our discussion in https://github.com/PX4/Devguide/issues/751 this is a new script for ROS melodic installation on Ubuntu 18.04. It is based on http://wiki.ros.org/melodic/Installation/Ubuntu and https://github.com/mavlink/mavros/tree/master/mavros#required-dependencies

Can you minimally check the mavros part is correct. 
The full build log is [here](https://gist.github.com/hamishwillee/2b32eb11d7cfad4efa1d92b2b8c8b6d6) - it looks good except for an error in RTPS build. 
I suspect that is a source rather than setup problem.

Basically I'm hoping you can tell me that this script is indeed OK, and then I can tidy up rest of docs.

Reply:
We should actually be installing Fast-RTPS 1.7.1 instead of 1.5.0.

Reply:
OK. I've updated (and tested) that script too.